### ISSUE TITLE
`public_member_api_docs` on Enum values having an annotation

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1396]
_From @cedx on December 31, 2018 10:45_

* Dart SDK: 2.1.0 (Tue Nov 13 18:22:02 2018 +0100) on "windows_x64"
* OS: Windows 10

I'm using the rule `public_member_api_docs` from the Dart linter.

If I use an annotation on a documented member of a class, everything is OK. But if I do the same thing with a enumerated value, the analyzer triggers an error: `Document all public members`.

```dart
// This class does not trigger the analyser warning.
class Foo {
  /// A comment.
  @JsonKey(defaultValue: '')
  String bar;
}

// This enum triggers the analyzer warning.
enum Baz {
   /// A comment.
  @JsonValue('contact')
  qux
}
```

Removing the annotation on the enumerated value also removes the analyzer warning.

_Copied from original issue: dart-lang/sdk#35527_

**Comment by ManuelDCR** [comment]
From my understanding problem still persists on Dart version `2.7.1`.

**Comment by pq** [comment]
Thanks for confirming @ManuelDCR and sorry for the inconvenience!

**Comment by pq** [comment]
Confirmed that it looks like the `EnumConstantDeclaration` AST node is returning a `null` `documentationComment` in this case.

@stereotype441 @scheglov: is this a known issue?

**Comment by stereotype441** [comment]
> Confirmed that it looks like the `EnumConstantDeclaration` AST node is returning a `null` `documentationComment` in this case.
> 
> @stereotype441 @scheglov: is this a known issue?

@pq I did a search through the issue tracker and couldn't find anything.  Would you mind filing an analyzer issue?

**Comment by pq** [comment]
@stereotype441, here ya go: https://github.com/dart-lang/sdk/issues/40865 👍

**Comment by srawlins** [comment]
The fix for the associated, blocking issue, has landed and been released.
